assets/js/data/cart/thunks.ts

```diff
@@ -41,6 +42,8 @@ export const receiveCart =
 			cartItemsPendingDelete: select.getItemsPendingDelete(),
 		} );
 		dispatch.setCartData( newCart );
+		maybeTrackCartPageView( newCart );
+		maybeTrackCheckoutPageView( newCart );
```

Comment:
I think this is the main issue I'm having is that I just don't think we should bake tracking logic into our app logic. We're now adding a direct link between our checkout logic and the tracking logic, is it possible for tracking to hook somehow externally into any events we might emit? if we're not emitting the correct events, we can consider that